CMakeLists.txt

```diff
@@ -15,6 +15,10 @@
 # We require at least CMake V3.3.2
 cmake_minimum_required(VERSION 3.3.2 FATAL_ERROR)
 
+if (HPX_WITH_CUDA AND CMAKE_VERSION VERSION_LESS 3.9)
```

Comment:
Isn't `HPX_WITH_CUDA` defined only further down in this file?

Reply:
yes, but the first time someone runs cmake, they will either use `cmake -DHPX_WITH_CUDA` on the command line, or they will run it once with `ccmake` then change the gui setting to enable `HPX_WITH_CUDA` and then when they configure, they'll get the error.